scipy/spatial/transform/rotation.py

```diff
@@ -281,3 +281,115 @@ def as_rotvec(self):
             return rotvec[0]
         else:
             return rotvec
+
+    @staticmethod
+    def _make_dcm_from_angle(char, angle, intrinsic=False):
+        sinx = np.sin(angle)
+        cosx = np.cos(angle)
+
+        axis = char.lower()
+        if axis == 'z':
+            dcm = np.array([
+                [cosx, -sinx, 0],
+                [sinx, cosx, 0],
+                [0, 0, 1]
+                ])
+        elif axis == 'y':
+            dcm = np.array([
+                [cosx, 0, sinx],
+                [0, 1, 0],
+                [-sinx, 0, cosx]
+                ])
+        elif axis == 'x':
+            dcm = np.array([
+                [1, 0, 0],
+                [0, cosx, -sinx],
+                [0, sinx, cosx]
+                ])
+
+        return dcm.transpose() if intrinsic else dcm
+
+    @classmethod
+    def _make_dcm_from_euler(cls, axis_str, angle):
+        extrinsic = recompile("^[xyz]{1,3}$")
+        intrinsic = recompile("^[XYZ]{1,3}$")
+
+        if len(axis_str) != angle.shape[0]:
+            raise ValueError("Number of axes specified in {0} does not match "
+                             "number of angles given in {1}".format(axis_str,
+                                                                    angle))
+        if ((extrinsic.match(axis_str) is None) and
+                (intrinsic.match(axis_str) is None)):
+                raise ValueError("Cannot allow mixing of intrinsic and "
+                                 "extrinsic rotations in sequence {}".format(
+                                    axis_str))
+
+        dcm = np.eye(3)
+        if extrinsic.match(axis_str) is not None:
+            # Build dcm with pre-multiplication
+            for ind, char in enumerate(axis_str):
+                dcm = np.einsum('...ij,...jk->...ik',
+                                cls._make_dcm_from_angle(char, angle[ind],
+                                                         False), dcm)
+        elif intrinsic.match(axis_str) is not None:
+            # Build dcm with post multiplication
+            for ind, char in enumerate(axis_str):
+                dcm = np.einsum('...ij,...ik->...ik', dcm,
+                                cls._make_dcm_from_angle(char, angle[ind],
+                                                         True))
+        return dcm
+
+    @classmethod
+    def from_euler(cls, seq, angle_mat, deg=False):
+        """Initialize rotation from Euler angles.
+
+        Parameters
+        ----------
+        seq : str or array of str
+            Each str[i] (or str itself) must be a string of upto 3 characters
```

Comment:
I vote for `seq` being a string only. Typically, if we want to initialize multiple rotations the sequence will be the same for all of them. This will simplify things greatly.